metals/src/main/scala/scala/meta/internal/metals/BloopInstall.scala

```diff
@@ -201,4 +212,38 @@ final class BloopInstall(
       }
   }
 
+  private def requestBuildToolChoice(
+      buildTools: List[BuildTool]
+  ): Future[Option[BuildTool]] = {
+    languageClient
+      .showMessageRequest(ChooseBuildTool.params(buildTools))
+      .asScala
+      .map { choice =>
+        val selectedBuildTool =
+          buildTools.find(buildTool =>
+            new MessageActionItem(buildTool.executableName) == choice
+          )
+        selectedBuildTool match {
+          case Some(buildTool) => {
+            val isCompatibleVersion = SemVer.isCompatibleVersion(
+              buildTool.minimumVersion,
+              buildTool.version
+            )
+            if (isCompatibleVersion) {
```

Comment:
This seems duplicated with `MetalsLanguageServer.supportedBuildTool`, if we filter them before, there will be no need for checking it again.